src/alire/alire-user_pins.adb

```diff
@@ -312,6 +322,9 @@ package body Alire.User_Pins is
                               else Origins.Short_Commit (+This.Commit))
          elsif Branch (This).Has_Element
          then "#" & TTY.Emph (+This.Branch)
+         else "")
+       & (if Subdir (This).Has_Element
```

Comment:
I don't think this is structured properly right now: it will produce something like `url/root.git#branch#subdir`. Probably not what we want.